components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/service/AppGatewayAdmin.java

```diff
@@ -0,0 +1,469 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.appmgt.gateway.service;
+
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.lang.StringEscapeUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.appmgt.gateway.dto.WebAppData;
+import org.wso2.carbon.appmgt.gateway.dto.ResourceData;
+import org.wso2.carbon.appmgt.api.AppManagementException;
+import org.wso2.carbon.appmgt.gateway.internal.ServiceReferenceHolder;
+import org.wso2.carbon.appmgt.gateway.utils.RESTAPIAdminClient;
+import org.wso2.carbon.core.AbstractAdmin;
+import org.wso2.carbon.rest.api.stub.types.carbon.APIData;
+import org.apache.axiom.om.impl.llom.util.AXIOMUtil;
+import org.wso2.carbon.sequences.common.SequenceEditorException;
+
+import javax.xml.stream.XMLStreamException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AppGatewayAdmin extends AbstractAdmin {
```

Comment:
Formatting issues and Java doc comment issues are fixed in https://github.com/wso2/carbon-appmgt/pull/539/files